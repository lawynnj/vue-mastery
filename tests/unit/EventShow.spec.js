import EventShow from '@/views/EventShow'
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('EventList.vue', () => {
  let actions, store

  beforeEach(() => {
    actions = {
      fetchEvent: jest.fn()
    }

    const event = {
      namespaced: true,
      actions,
      state: {
        event: {}
      }
    }
    store = new Vuex.Store({
      modules: {
        event
      }
    })
  })

  it('dispatches "add" when it EventList is created', () => {
    shallowMount(EventShow, { store, localVue })
    expect(actions.fetchEvent).toHaveBeenCalled()
  })

  it('EventList should render correctly', () => {
    const wrapper = shallowMount(EventShow, { store, localVue })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
