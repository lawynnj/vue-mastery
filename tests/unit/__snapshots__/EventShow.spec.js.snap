// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList.vue EventList should render correctly 1`] = `
<div
  class="event-show"
>
  <div
    class="event-header"
  >
    <span
      class="eyebrow"
    >
      @ on 
    </span>
     
    <h1
      class="title"
    >
      
    </h1>
     
    <h5>
      Organized by 
    </h5>
     
    <h5>
      Category: 
    </h5>
  </div>
   
  <baseicon
    name="map"
  >
    <h2>
      Location
    </h2>
  </baseicon>
   
  <address>
    
  </address>
   
  <h2>
    Event details
  </h2>
   
  <p>
    
  </p>
   
  <h2>
    
    Attendees
    
    <span
      class="badge -fill-gradient"
    >
      0
    </span>
  </h2>
   
  <ul
    class="list-group"
  />
</div>
`;
